source/fundamentals/connection/connection-options.txt

```diff
@@ -0,0 +1,418 @@
+its properties, and pass it as an argument to the ``MongoClient`` constructor:
+
+.. literalinclude:: /includes/fundamentals/code-examples/connection/MongoClientSettingsConfig.cs
+   :language: csharp
+   :dedent:
+
+Connection Options
+------------------
+
+The following table lists each connection option available in the
+``MongoClientSettings`` class and, if possible, how to achieve the same result in 
+the connection string. If a ``MongoClientSettings`` property maps to more than one 
+option in the connection string, the **Connection URI Example** column shows all 
+relevant options. 
+
+.. note::
+
+  If you're using a query parameter for a time duration, the value must be in 
+  milliseconds. For example, to specify 60 seconds, use the value ``60000``. If you're 
+  using a ``MongoClientSettings`` object for a time duration, use the appropriate 
+  ``TimeSpan`` value.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 40 30 30
+
+   * - **MongoClientSettings** Property
+     - Connection URI Example
+     - Description
+
+   * - **AllowInsecureTls**
+     - ``tlsInsecure=true``
+     - | Specifies whether to relax TLS constraints as much as possible. This can include 
+         allowing invalid certificates or hostname mismatches.
+       |
+       | **Data Type**: {+bool-data-type+}
+       | **Default**: ``false``
+
+   * - **ApplicationName**
+     - ``appName=yourApp``
+     - | The app name the driver passes to the server in the client metadata as part of 
+         the connection handshake. The server prints this value to the MongoDB logs once 
+         the connection is established. The value is also recorded in the slow query logs 
+         and profile collections.
+       |
+       | **Data Type**: {+string-data-type+}
+       | **Default**: ``null``
+
+   * - **AutoEncryptionOptions**
+     - {+not-available+}
+     - | Settings for automatic client-side encryption.
+       |
+       | **Data Type**: ``AutoEncryptionOptions``
+       | **Default**: ``null``
+
+   * - **ClusterConfigurator**
+     - {+not-available+}
+     - | Low-level configuration options for sockets, TLS, cluster, and others. 
+       | 
+       | **Data Type**: ``Action<ClusterBuilder>``
+       | **Default**: ``null``
+
+   * - **Compressors**
+     - ``compressors=snappy,zstd``
+     - | The preferred compression types, in order, for wire-protocol messages sent to 
+         or received from the server. The driver uses the first of these compression types 
+         that the server supports. See :ref:`csharp-network-compression` for 
+         more information. 
+       | 
+       | **Data Type**: ``CompressorConfiguration``
+       | **Default**: ``null``
+
+   * - **ConnectTimeout**
+     - ``connectTimeoutMS=0``
+     - | The length of time the driver tries to establish a single TCP socket connection 
+         to the server before timing out. Specifying ``0`` disables the connection timeout.
```

Comment:
Do you mean setting `connectTimeout=TimeSpan.Zero` will actually set the timeout to 30 seconds in the driver? Or do you mean that the writing incorrectly suggests that it will?